components/board/genericlinux/hw_pwm.go

```diff
@@ -0,0 +1,174 @@
+// Package genericlinux is for Linux boards. This particular file is for using sysfs to
+// interact with PWM devices. All of these functions are idempotent: you can double-export a pin or
+// double-close it with no problems.
+package genericlinux
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/pkg/errors"
+)
+
+type pwmDevice struct {
+	// These values are immutable
+	chipPath string
+	line     int
+
+	mu sync.Mutex
+
+	// These values are mutable
+	periodNs         uint64
+	activeDurationNs uint64
+	isExported       bool
+	isEnabled        bool
+}
+
+func newPwmDevice(chipName string, line int) (*pwmDevice, error) {
+	// There should be a single directory within /sys/devices/platform/<chipName>/pwm/, whose name
+	// is mirrored in /sys/class/pwm. That's the one we want to use.
+	chipDir := fmt.Sprintf("/sys/devices/platform/%s/pwm", chipName)
```

Comment:
:flushed: I checked on some other boards:
- On a BeagleBone, and they're in `/sys/devices/platform/bus@100000/<chipname>/pwm` instead! There's that extra `bus@100000/` that the Orin is missing.
- On an Up-4000, they're in `/sys/devices/pci0000:00/<chipname>/pwm`. This time, the `platform/` has been replaced with `pci0000:00/` (though `/sys/devices/platform` also exists; that's just not where the PWM chip is mentioned).
- My Framework laptop doesn't have user-accessible PWM at all.
- The Raspberry Pi, as we know, also doesn't have a hardware PWM chip.

We could add the extra directories into the relevant data.go files and just pretend that the chip name has a slash in the middle, but that feels a little bit wrong to me (there appears to be this standard data format that we'd be using in a nonstandard way). 

This certainly needs to be changed eventually, but I don't have a sense of how much effort it will be to find the universally correct approach given the 3 data points I have so far. I'll create a Jira ticket, add a TODO here, and look into this further as we generalize to other boards.

Reply:
There's a similar pattern for checking for this in `data.go` for gpio pins using sysfs. Starts at this [line](https://github.com/viamrobotics/rdk/blob/99e2f64add4c14fa40414c9903d6fb572efbfc21/components/board/genericlinux/data.go#L94).

Ideally, we'd want all looking up of directories to happen in `data.go` and all installation of functionality to happen in the files with names themselves, but not blocking on that. It just will make migrating the other boards to ioctl and using this hardware pwm a bit easier for future us.

Reply:
I agree with making a TODO and accompanying JIRA ticket